### ISSUE TITLE
Higher order effects bug

### ISSUE THREAD

**Issue by Simre1** [isovector/polysemy#57]
There seems to be a bug with higher order effects. Here is an example with the current version of polysemy on hackage (0.1.2.1) and ghc-8.6.4.

```
main :: IO ()
main = runM . runReader 4 $ test

test :: Sem '[Reader Int, Lift IO] ()
test = do
  ask @Int >>= liftIO . print
  local (+(4 :: Int)) $ do
    ask @Int >>= liftIO . print
```
Executing the main function, I would expect:
`4`
`8`

What's actually happening is:
`4`
`4`

**Comment by isovector** [comment]
What an embarrassing bug! I can duplicate it, but not entirely sure _why_. I'll dig in tomorrow!

**Comment by mitchellwrosen** [comment]
Randomly chiming in here, do not be embarrassed, software is reeeeeally hard... ;)

**Comment by isovector** [comment]
Thanks for finding this --- it's a pretty duplicitous bug in the fundamental combinators. Essentially they'd handle the higher-order effects automatically using the currently-run interpreter, but then would call `lift` on it so it seemed like you still had to handle that thing. The behavior "works" for any interpreter that just wants to call itself, but `runReader` isn't in that class.

**Comment by isovector** [comment]
I've released the fix as version 0.2.0.0, along with deprecating the other versions on hackage. Thanks again for the bug report!

**Comment by Simre1** [comment]
Thanks for fixing the problem so quickly! It works perfectly now.
